apis/logging/v1/output_types.go

```diff
@@ -276,6 +276,21 @@ type Splunk struct {
 	// Should be a valid JSON object
 	// +optional
 	Fields []string `json:"fields,omitempty"`
+
+	// IndexKey is a meta-data key field to use to send events to.
+	// For example: 'IndexKey: kubernetes.namespace_name` will use the kubernetes
+	// namespace as the index.
+	// If the IndexKey is not found, the `log_type` will be used as the index.
```

Comment:
Based upon the vector's [statement](https://vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#index) regarding "default", we need to walk back our intent to use "log_type" here.  This would break in an upgrade scenario

Reply:
Currently if no `index` is specified, 
> the default index defined within Splunk is used.

The `default` index in Splunk is `main`. What I can do here is fallback to `main` as the index if the specified key isn't found. Will that work?

Reply:
@Clee2691 I would leave it unset completely since it appears to me you can define the default index on the splunk side.  Setting it to `main` on the client locks us in and does not meet backwards compatibility since we never set it previously

Reply:
After some testing, it seems that I can set the index as an empty string (`""`) which will allow logs to be sent to the default index. 

This is because if the key is not present in the log record and nothing is set, vector complains with:
```
Failed to render template for "index". error=Missing fields on event: ["write_index"] error_type="template_failed" stage="processing" internal_log_rate_limit=true
```
since I am using an artificial field, `.write_index`, to index logs.

Reply:
Do we need to update the comment to be backwards compatible?

Reply:
Yes good catch